core/schema/modulesource.go

```diff
@@ -3028,7 +3093,32 @@ func applyArgumentConfigToFunction(fn *core.Function, argConfigs []*modules.Modu
 			if strings.EqualFold(arg.Name, argCfg.Name) || strings.EqualFold(arg.OriginalName, argCfg.Name) {
 				// Apply default value if specified
 				if argCfg.Default != "" {
-					arg.DefaultValue = core.JSON([]byte(fmt.Sprintf("%q", argCfg.Default)))
+					// Parse the default value as JSON to determine its type
+					var defaultValue any
+					if err := json.Unmarshal([]byte(argCfg.Default), &defaultValue); err != nil {
+						// If it's not valid JSON, treat it as a string literal
+						defaultValue = argCfg.Default
+					}
+
+					// Validate the type matches the argument type
+					if err := validateDefaultValueType(defaultValue, arg.TypeDef); err != nil {
```

Comment:
I think you could just do `val, err := typeDef.ToInput().Decoder().DecodeInput(defaultValue)` and remove `validateDefaultValueType`

Reply:
cc @kpenfound 👆